freqtrade/optimize/backtesting.py

```diff
@@ -252,8 +251,17 @@ def _get_sell_trade_entry(self, trade: LocalTrade, sell_row: Tuple) -> Optional[
         sell = self.strategy.should_sell(trade, sell_row[OPEN_IDX],  # type: ignore
                                          sell_row[DATE_IDX], sell_row[BUY_IDX], sell_row[SELL_IDX],
                                          low=sell_row[LOW_IDX], high=sell_row[HIGH_IDX])
-        if sell.sell_flag:
 
+        time_in_force = self.strategy.order_time_in_force['sell']
+        # confirm_trade_exit
+        if not strategy_safe_wrapper(self.strategy.confirm_trade_exit, default_retval=True)(
```

Comment:
Did you do some performance testing with / without this change?

i do have my concerns about performance with this - which is why i intentionally didn't add this functionality to backtesting in the first place.

This can be especially relevant for strategies with a lot of trades (a few 1000 over a year), small candle sizes (5m) - and relatively long holding times - as then this function is HAMMERED (the sell function alone will be called once for every candle while a trade is open - now multiply this with the number of pairs and trades you open, and it'll be a problem).

In backtesting, every function call added can have a big performance impact (which implicitly also impacts hyperopt, so we need to be very careful with this), and why protections need to be explicitly enabled.


Can you please perform some benchmarking of this change:
A simple test can be using time (`time freqtrade backtesting ....`) - and comparing this vs. the underlying develop branch performance (best on a strategy that doesn't implement `confirm_*` methods).
Please use a strategy that does at least 2000 trades over a 1 year period on 5m candles, and post the timings before / after this change.

Reply:
I've used it for more than a week and didn't recognized any performance issues.
Sure I will benchmark it and share the results here.

Reply:
Super thanks!

also, If we now use this during backtesting / hyperopt, it should be documendted in `bot-basics.md` (which documents the flow).

Reply:
> Please let me know if it's fine
Also, I would like you to check the code and make sure the parameters i've passed to the methods are correct (as I'm not that familiar with the project)

don't worry - that's DEFINITELY part of the review ... not because you're new or not, but because typos can happen easily.